### PR TITLE
Add convenience options runtests

### PR DESCRIPTION
**Pull request by hjmjohnson**
Related to #476, enable mypy static tests

### Description
The mypy static tests are still the type consistency standard that other static analysis tools are compared. Add this static checker to the default testing.

### Status
Ready

### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New tests added to cover the changes
- [X] Docstrings/Documentation updated

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@wyli This adds a more expansive mypy testing, and also improves the runtest.sh script.

File: .github/workflows/pythonapp.yml [hunk 1]
```diff
@@ -26,6 +27,10 @@ jobs:
         flake8 --version
         python -c 'import monai; monai.config.print_config()'
         MYPYPATH=$(pwd)/monai flake8 $(pwd) --count --statistics
+    - name: Type check with mypy statc analysis
+      run: |
+        mypy --version
```
Comment (wyli) [inline .github/workflows/pythonapp.yml:32]: do we need to add mypy to requirements-dev.txt?

Comment (hjmjohnson) [reply .github/workflows/pythonapp.yml:32]: It is included by transitivity from flake8-mypy, but we should probably include it anyway.

**Comment by hjmjohnson** [comment]
@wyli Let's hold off on this till tomorrow.  I found a way to integrate it with the CI environment better so that we have only one place to manage the command lines for the linting tools.

It will require a little more scripting, but then one definition can be used for both CI and local testing.

**Comment by hjmjohnson** [comment]
@wyli Here is the much-improved version.  

Now the CI uses command-line options that end-users have access too: 
https://github.com/Project-MONAI/MONAI/pull/496/commits/138b5369585c490edd18d178209826b66695a067#diff-7e3bc64c9dd0eb02c0cac6101ee4e286R29

**Pull request merged**